js/tuxexchange.js

```diff
@@ -0,0 +1,533 @@
+            'last': this.safeFloat (tickerData, 'last'),
+            'high': this.safeFloat (tickerData, 'high24hr'),
+            'low': this.safeFloat (tickerData, 'low24hr'),
+            'percentage': this.safeFloat (tickerData, 'percentChange'),
+            'baseVolume': this.safeFloat (tickerData, 'baseVolume'),
+            'quoteVolume': this.safeFloat (tickerData, 'quoteVolume'),
+            'bid': this.safeFloat (tickerData, 'highestBid'),
+            'ask': this.safeFloat (tickerData, 'lowestAsk'),
+            'bidVolume': undefined,
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': undefined,
+            'previousClose': undefined,
+            'change': undefined,
+            'average': undefined,
+            'info': tickerData,
+        };
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOrderBook () requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const codes = this.getIdsFromSymbol (symbol);
+        if (codes['base'] !== 'BTC') {
+            throw new NotSupported (this.id + ' this exchange only trades on symbols with BTC as base');
+        }
+        if (limit !== undefined) {
+            // At some point it would be worthwhile to extend the base parseOrderBook to include a limit
+            throw new NotSupported (this.id + ' fetchOrderBook () does not support a "limit" argument for this exchange');
+        }
+        let orderBookRequest = {
+            'coin': this.currencyId (codes['quote']),
+            'method': 'getorders',
+        };
+        const orderBook = await this.publicGet (this.extend (orderBookRequest, params));
+        const result = this.parseOrderBook (orderBook);
+        return result;
+    }
+
+    parseTrade (trade, market) {
+        // The API exposees different fields for the gettrades and getmytradehistory endpoints
+        const isPrivateTrade = trade['orderId'] !== undefined;
+        const tradeDate = this.parseDate (this.safeString (trade, 'date'));
+        const timestamp = tradeDate.valueOf ();
+        let priceString = undefined;
+        if (isPrivateTrade) {
+            priceString = this.safeString (trade, 'price');
+        } else {
+            priceString = this.safeString (trade, 'rate');
+        }
+        const parsedTrade = {
+            'id': this.safeString (trade, 'tradeid'),
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': market.id,
+            'type': 'limit',
+            'side': this.safeString (trade, 'type'),
+            'price': this.asFloat (priceString),
+            'amount': this.asFloat (this.safeString (trade, 'amount')),
+            'cost': this.asFloat (this.safeString (trade, 'total')),
+        };
+        if (isPrivateTrade) {
```

Comment:
The `fields` of a unified trade structure should not depend on whether it's a private or a public trade – we should include all fields, including fees and other data, and the unknown values should be `undefined`.

Reply:
See the requirements for all unified data structures here: https://github.com/ccxt/ccxt/wiki/Requirements